com.unity.netcode.gameobjects/Tests/Runtime/Metrics/MetricsDispatchTests.cs

```diff
@@ -0,0 +1,122 @@
+#if MULTIPLAYER_TOOLS
+using System;
+using System.Collections;
+using NUnit.Framework;
+using Unity.Multiplayer.Tools.NetStats;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Unity.Netcode.RuntimeTests.Metrics
+{
+    public class MetricsDispatchTests
+    {
+        private class MockMetricsObserver : IMetricObserver
+        {
+            public Action OnObserve;
+            public void Observe(MetricCollection collection) => OnObserve?.Invoke();
+        }
+
+        private int m_NumTicks;
+        private int m_NumDispatches;
+
+        private NetworkManager m_NetworkManager;
+        private NetworkTimeSystem timeSystem => m_NetworkManager.NetworkTimeSystem;
+        private NetworkTickSystem tickSystem => m_NetworkManager.NetworkTickSystem;
+        private uint m_TickRate = 1;
+
+        [SetUp]
+        public void SetUp()
+        {
+            Assert.IsTrue(NetworkManagerHelper.StartNetworkManager(
+                out m_NetworkManager,
+                NetworkManagerHelper.NetworkManagerOperatingMode.Host,
+                new NetworkConfig()
+            {
+                TickRate = m_TickRate
+            }));
+
+            InitNetworkManager();
+
+            var networkMetrics = m_NetworkManager.NetworkMetrics as NetworkMetrics;
+            networkMetrics.Dispatcher.RegisterObserver(new MockMetricsObserver()
+            {
+                OnObserve = ()=> m_NumDispatches++
+            });
+        }
+
+        private void InitNetworkManager()
+        {
+            tickSystem.Tick += ()=> m_NumTicks++;
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            NetworkManagerHelper.ShutdownNetworkManager();
+            m_NumTicks = default;
+            m_NumDispatches = default;
+            m_NetworkManager = default;
+        }
+
+        [UnityTest]
+        public IEnumerator GivenMultipleTicks_OneDispatchOccurs()
+        {
+            AdvanceTicks(2);
+
+            // Wait one frame so dispatch occurs
+            yield return null;
+
+            Assert.AreEqual(2, m_NumTicks);
+            Assert.AreEqual(1, m_NumDispatches);
+        }
+
+        [UnityTest]
+        public IEnumerator GivenOneTick_OneDispatchOccurs()
+        {
+            AdvanceTicks(1);
+
+            // Wait one frame so dispatch occurs
+            yield return null;
+
+            Assert.AreEqual(1, m_NumTicks);
+            Assert.AreEqual(1, m_NumDispatches);
+        }
+
+        [UnityTest]
+        public IEnumerator GivenZeroTicks_ZeroDispatchesOccur()
+        {
+            yield return null;
+
+            Assert.AreEqual(0, m_NumTicks);
+            Assert.AreEqual(0, m_NumDispatches);
+        }
+
+        [UnityTest]
+        public IEnumerator GivenReinitializedNetworkManagerAfterOneTickExecuted_WhenFirstTickExecuted_MetricsDispatch()
```

Comment:
Sounds good. I think this is actually no longer an edge case after changing the implementation to a dirty flag. But I left the test in because the case should still pass even if the edge case isn't there. Added comment to reflect all this.